File: examples/python/set_output.py
import requests

API_URL = "http://localhost:3030"

def set_output(output_value):
    """Send a request to set the output value on the server."""
    url = f"{API_URL}/sp1/set-output"

    output_value = output_value.encode() if isinstance(output_value, str) else output_value
    
    data = {"output": list(output_value)}
    response = requests.post(url, json=data)
    return response.json()

if __name__ == "__main__":
    set_secret_result = set_output("cca")

    print(f"Set Secret Result: {set_secret_result}")

File: examples/javascript/setOutput.js
import axios from "axios";

const API_URL = 'http://localhost:3030';

function setOutput(outputValue) {
    // Convert the string to a byte array if outputValue is a string
    const outputArray = typeof outputValue === 'string'
        ? Array.from(Buffer.from(outputValue, 'utf-8'))
        : outputValue;

    const url = `${API_URL}/sp1/set-output`;
    const data = { output: outputArray };

    return axios.post(url, data)
        .then(response => response.data)
        .catch(error => {
            console.error('Error:', error);
            throw error;
        });
}

(async () => {
    try {
        const setSecretResult = await setOutput("cca");
        console.log(`Set Secret Result: ${JSON.stringify(setSecretResult)}`);
    } catch (error) {
        console.error('Failed to set output:', error);
    }
})();
